### PR TITLE
Create signed cookies lambda

### PR DESCRIPTION
**Pull request by MancunianSam**
Add signed cookies lambda.

This is equivalent to the lambda in tdr-sign-cookies but is written in Python.
A quick test on integration shows that for a cold start, the response time is just over a second compared to 10 seconds for the Scala lambda.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,35 @@
...
\ No newline at end of file
+# TDR Signed Cookies
+
+This lambda provides the response when the front end makes a request to upload.tdr.nationalarchives.gov.uk. The request goes:
+
+Route53 -> Cloudfront -> API Gateway -> Lambda
+
+The lambda carries out the following steps.
+
+1. Decode the incoming json from API gateway and retrieve the access token.
```
Comment (TomJKing) [inline README.md]: Couple of typos: `Decodes the incoming ... and retrieves ...`

File: README.md [hunk 2]
```diff
@@ -1 +1,35 @@
...
+2. Validate the token with the jwt library. If this has expired or is otherwise invalid, the lambda returns 401.
```
Comment (TomJKing) [inline README.md]: Typo: `Validates the token ...`

File: README.md [hunk 3]
```diff
@@ -1 +1,35 @@
...
+3. Create the signed cookies using the private key from the environment variables.
```
Comment (TomJKing) [inline README.md]: Typo: `Creates the ...`

File: README.md [hunk 4]
```diff
@@ -1 +1,35 @@
...
+4. Create the response with the correct `Set-Cookies` headers set
```
Comment (TomJKing) [inline README.md]: Typo: `Creates the ...`

File: README.md [hunk 5]
```diff
@@ -1 +1,35 @@
...
+5. Set the `Access-Control-Allow-Origin` header based on the environment and origin header.
```
Comment (TomJKing) [inline README.md]: Typo: `Sets the ...`

File: README.md [hunk 6]
```diff
@@ -1 +1,35 @@
...
+6. Return a 200 response.
```
Comment (TomJKing) [inline README.md]: Typo: `Returns a ...`

File: README.md [hunk 7]
```diff
@@ -1 +1,35 @@
...
+
+The docker image is built from `Dockerfile` in the root of the project and the function is built using the `build-dependencies.sh` script.
+
+This is built using GitHub actions in the `.github/workflows/build.yml` file.
+
+## Running Locally
+There is a `signed_cookies_runner` file with some example json. To run this, you will need to set the environment variables to the same values as the integration lambda. You can get these by running this command with integration credentials
+
+`aws lambda get-function --function-name tdr-sign-cookies-intg --query  'Configuration.Environment.Variables'  
+`
```
Comment (TomJKing) [inline README.md]: Extra line breaks.

File: README.md [hunk 8]
```diff
@@ -1 +1,35 @@
-# TDR Signed Cookies
\ No newline at end of file
+# TDR Signed Cookies
```
Comment (TomJKing) [inline README.md:1]: Could you add a section about running the test locally?

**Review by TomJKing — COMMENTED** [review]
Looks good (assuming it works ;) ). Just some very minor comments on the `README`

There appear to be a couple of empty files include, not sure they should be there:

- `src/__init__.py`
- `test/utils/__init__.py`

**Comment by MancunianSam** [comment]
> There appear to be a couple of empty files include, not sure they should be there:
> 
> * `src/__init__.py`
> * `test/utils/__init__.py`

They are needed, they make it a Python module apparently which is needed for the tests to run. I tried it without them and it didn't work.

File: README.md (hunk 8 above)
Comment (MancunianSam) [reply README.md:1]: I've added in a section about running in the terminal and in PyCharm/IntelliJ

**Comment by TomJKing** [comment]
@MancunianSam Did you push any changes for this PR?

**Comment by MancunianSam** [comment]
I didn't, sorry. I've just committed and pushed them.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**